### PR TITLE
Add resources to worker and jobs

### LINKED ISSUE #701: Resources in worker and job pods
As far as I know, there are no way of setting the container resources for both worker and jobs pods.

This issue is intended to discuss the way of implement this to create a PR with an agreed schema.

**Actual status**

| pod | container | status |
|---|---|---|
| worker | init-container | no resources |
| worker | worker container | no resources |
| job | init-container | no resources |
| job | job container | set by job configuration |

**Desired status**: ability to set resources in every container

**Proposal**

| pod | container | status |
|---|---|---|
| worker | init-container | set by project definition |
| worker | worker container | set by flags in the controller |
| job | init-container | set by project definition |
| job | job container | set by job configuration |

I'd create four flags for the worker configuration:
- `worker-limits-cpu`
- `worker-limits-memory`
- `worker-requests-cpu`
- `worker-requests-memory`

This would set the resources on the main container of the worker.

For the `vcs-sidecar` init-containers, I'd set the resources in the project configuration:
- `vcsSidecar.limits.cpu`
- `vcsSidecar.limits.memory`
- `vcsSidecar.requests.cpu`
- `vcsSidecar.requests.memory`

What do you think about this?

### PR DESCRIPTION
**Pull request by jchanam**
Implements #701 

Adds flags for the controller to set the worker container resources.

Adds project configuration to set the init-container vcs-sidecar resources.

### CODE REVIEW

File: charts/brigade-project/templates/secret.yaml [hunk 1]
```diff
@@ -28,6 +28,10 @@ stringData:
   {{- else }}
   vcsSidecar: {{ printf "deis/git-sidecar:v%s" .Chart.Version }}
   {{- end }}
+  vcsSidecarResources.limits.cpu: {{ default "" .Values.vcsSidecarResources.limits.cpu | quote }}
```
Comment (vdice) [inline charts/brigade-project/templates/secret.yaml]: I believe this section needs to be wrapped in a `  {{ if .Values.vcsSidecarResources }}...{{- end}}` block to satisfy the chart linter (`make test-chart` currently throws an error)

File: brigade-controller/cmd/brigade-controller/main.go [hunk 2]
```diff
@@ -31,6 +31,10 @@ func main() {
 	flag.StringVar(&ctrConfig.WorkerImage, "worker-image", defaultWorkerImage(), "kubernetes worker image")
 	flag.StringVar(&ctrConfig.WorkerPullPolicy, "worker-pull-policy", defaultWorkerPullPolicy(), "kubernetes worker image pull policy")
 	flag.StringVar(&ctrConfig.WorkerServiceAccount, "worker-service-account", defaultWorkerServiceAccount(), "kubernetes worker service account name")
+	flag.StringVar(&ctrConfig.WorkerRequestsCPU, "worker-requests-cpu", "", "kubernetes worker cpu requests")
```
Comment (vdice) [inline brigade-controller/cmd/brigade-controller/main.go]: Shall we add/expose this configuration in the brigade chart such that these values can be set in the chart values.yaml?

**Comment by jchanam** [comment]
@vdice Thanks for the comments! I've done what you suggested

File: charts/brigade-project/Chart.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
 apiVersion: v1
 description: Create a Brigade project
 name: brigade-project
-version: 0.18.0
+version: 0.18.1
```
Comment (vdice) [inline charts/brigade-project/Chart.yaml]: It looks like this value is used for sidecar image tags, which don't exist, naturally
```
  Normal   BackOff    13s   kubelet, aks-nodepool1-26912603-0  Back-off pulling image "deis/git-sidecar:v0.18.1"
  Warning  Failed     13s   kubelet, aks-nodepool1-26912603-0  Error: ImagePullBackOff
```
I say we bump this back to `0.18.0` here and in the `brigade/Chart.yaml` and we'll just bump to the next tagged version once this changeset is included in a release.

File: charts/brigade/values.yaml [hunk 4]
```diff
@@ -45,6 +45,11 @@ controller:
   # The default tag is the value of .Chart.AppVersion
   #tag:
   #pullPolicy: IfNotPresent
+  #workerResources:
+  #  limitsCPU:
```
Comment (vdice) [inline charts/brigade/values.yaml]: nit: do you think the structure of these values could be aligned with those used for the vcsSidecarResources in the brigade-project values.yaml?

(Meaning, `workerResources.limits.cpu`, `workerResources.limits.memory`, etc.)

File: charts/brigade-project/Chart.yaml (hunk 3 above)
Comment (jchanam) [reply charts/brigade-project/Chart.yaml]: I've added the `AppVersion` to be used for the image tag, in order to allow the chart to evolve.

This is the way it's done on the brigade chart, also It's what I understand is the best way to do it, after reading the helm documentation: https://docs.helm.sh/developing_charts/#charts-and-versioning and https://docs.helm.sh/developing_charts/#the-appversion-field

**Comment by vdice** [comment]
This looks great; thanks again @jchanam !

**Pull request merged**